### PR TITLE
Proposal to add OCI Image Annotations on Buildpacks

### PR DESCRIPTION
**Pull request by candrews**
## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [ ] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [ ] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: text/0000-oci-image-annotations.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+All Paketo Buildpacks must have at least the following OCI image annotations with values as defined by the [OCI Image Format Specification's Pre-Defined Annotation Keys](https://github.com/opencontainers/image-spec/blob/v1.1.0/annotations.md#pre-defined-annotation-keys):
+* `org.opencontainers.image.source`
+* `org.opencontainers.image.revision`
+* `org.opencontainers.image.title`
+* `org.opencontainers.image.version`
+
+The following additional annotation are recommended:
+* `org.opencontainers.image.url`
+* `org.opencontainers.image.description`
```
Comment (sophiewigmore) [inline text/0000-oci-image-annotations.md:13]: according to the spec you sent, would `org.opencontainers.image.created`, `org.opencontainers.image.authors`, `org.opencontainers.image.documentation`, etc be required as well?

Comment (candrews) [reply text/0000-oci-image-annotations.md:13]: I don't believe they would be required, but if they are available, they should be specified. For paketo, I don't see any of that information readily available in `buildpack.toml` or elsewhere, so I didn't include them in this RFC.

Comment (sophiewigmore) [reply text/0000-oci-image-annotations.md:13]: How do you know which fields in the upstream spec you sent are required?

Comment (candrews) [reply text/0000-oci-image-annotations.md:13]: I'm assuming that no fields are required, only that if they are present, they are required to meet the requirements for that field's value.

That's what I see in the wild, at least. I can't find a single image that specifies all of those fields, but I see a number that specify some.

Comment (sophiewigmore) [reply text/0000-oci-image-annotations.md:13]: Ah ok. The wording of the RFC `All Paketo Buildpacks must have at least the following OCI image annotations` made me think that this was required to be OCI compliant which was why I was confused.

I think this RFC seems like a good idea, especially if we can pull from the buildpack.toml fields as you suggested

**Comment by ForestEckhardt** [comment]
As a quick clarifying question this is seeking to add these image labels to the buildpack images we release. Not to add them to the images that are produced by the buildpacks correct?

**Comment by candrews** [comment]
> As a quick clarifying question this is seeking to add these image labels to the buildpack images we release. Not to add them to the images that are produced by the buildpacks correct?

Correct.

Although I do think that's a good idea, I think it would best be pursued in a separate RFC. https://github.com/paketo-buildpacks/image-labels does most of that already, so the implementation of such an RFC would be to apply it by default with reasonable default values.

**Comment by candrews** [comment]
> It this is something that you would like added to the buildpack images themselves that I think it my make sense to try and push this up the stack into `pack`. We use the `pack buildpack package` to turn a "formatted" tarball into an OCI image so it may make sense to ask them to add these terms to that tarball by default using the values present in the `buildpack.toml`.

That makes sense to me! That would benefit more than just paketo too, which is nice.


> I would be more than happy to help facilitate an interaction if that is something that you are interested in.

I'm certainly interested, can you please point me in the right direction? Thank you!

**Comment by ForestEckhardt** [comment]
@candrews Is there anything else that needs to be done on this RFC? Should we close it out in favor of an upstream PR or do you want to modify this one?

**Comment by candrews** [comment]
> @candrews Is there anything else that needs to be done on this RFC? Should we close it out in favor of an upstream PR or do you want to modify this one?

I submitted the RFC to buildpacks at https://github.com/buildpacks/rfcs/pull/314 - could you please facilitate this process?

**Comment by ForestEckhardt** [comment]
Hey there @candrews,
I see that the upstream proposal was accepted. Is there anything that still needs to happen with this RFC or has everything you wanted been accomplished in the upstream RFC?

**Comment by candrews** [comment]
I believe the upstream RFC covers it. Now I hope it gets implemented 🤞 

Thank you!

**Pull request closed without merging**